iocage/cli/fetch.py

```diff
@@ -38,58 +44,113 @@ def validate_count(ctx, param, value):
+#                   "that location.\nDeveloper option, most will prefer to "
+#                   "use --plugins.")
+# @click.option("--plugins", help="List all available plugins for creation.",
+#              is_flag=True)
+# @click.argument("props", nargs=-1)
 @click.option("--update/--noupdate", "-U/-NU", default=True,
               help="Decide whether or not to update the fetch to the latest "
                    "patch level.")
-@click.option("--eol/--noeol", "-E/-NE", default=True,
-              help="Enable or disable EOL checking with upstream.")
-@click.option("--name", "-n", help="Supply a plugin name for --plugins to "
-                                   "fetch or use a autocompleted filename"
-                                   " for --plugin-file.\nAlso accepts full"
-                                   " path for --plugin-file.")
-@click.option("--accept/--noaccept", default=False,
-              help="Accept the plugin's LICENSE agreement.")
+# @click.option("--eol/--noeol", "-E/-NE", default=True,
+#               help="Enable or disable EOL checking with upstream.")
+# @click.option("--name", "-n", help="Supply a plugin name for --plugins to "
+#                                    "fetch or use a autocompleted filename"
+#                                    " for --plugin-file.\nAlso accepts full"
+#                                    " path for --plugin-file.")
+# @click.option("--accept/--noaccept", default=False,
+#              help="Accept the plugin's LICENSE agreement.")
+# Compat
+@click.option("--http", "-h", default=False,
+              help="Have --server define a HTTP server instead.", is_flag=True)
+# Compat files
+@click.option("--files", multiple=True,
+              help="Specify the files to fetch from the mirror. "
+              "(Deprecared: renamed to --file)")
+# def cli(url, files, release, update):
 def cli(**kwargs):
-    """CLI command that calls fetch_release()"""
-    ioc.IOCage().fetch(**kwargs)
+
+    release_input = kwargs["release"]
+
+    if isinstance(release_input, int):
+        try:
+            release = host.distribution.releases[int(release_input)]
+        except:
+            logger.error("Selection {release_input} out of range")
+            sys.exit(1)
+
+    try:
+        release = Release.Release(
+            name=release_input,
+            host=host,
+            logger=logger
+        )
+    except:
+        self.logger.error("Invalid Release '{release_input}'")
+
+    url_or_files_selected = False
+
+    # optional --url
+    try:
+        url = kwargs["url"]
+        if url:
+            release.mirror_url = url
+            url_or_files_selected = True
+    except:
+        pass
+
+    # optional --files
+    try:
+        files = kwargs["files"]
+        if files:
+            release.assets = list(files)
+            url_or_files_selected = True
+    except:
+        pass
+
+    if (url_or_files_selected is False) and (release.available is False):
```

Comment:
@skarekrow it's the accent of JavaScript shining through. An if is nothing without it's braces. In this case I was using the braces for better readability. Different languages may have different operator weights, so that I think being explicit makes it clear. Of course I can change it if you like.